src/OrchardCore.Modules/OrchardCore.Lists/GraphQL/ContainedPartContentTypeBuilder.cs

```diff
@@ -0,0 +1,33 @@
+using GraphQL.Types;
+using OrchardCore.ContentManagement.GraphQL.Queries.Types;
+using OrchardCore.ContentManagement.Metadata.Models;
+using OrchardCore.Lists.Models;
+
+namespace OrchardCore.Lists.GraphQL;
+
+internal sealed class ContainedPartContentTypeBuilder : IContentTypeBuilder
+{
+    public void Build(ISchema schema, FieldType contentQuery, ContentTypeDefinition contentTypeDefinition, ContentItemType contentItemType)
+    {
+        foreach (var listPart in contentTypeDefinition.Parts.Where(p => p.PartDefinition.Name.Equals(nameof(ListPart), StringComparison.OrdinalIgnoreCase)))
+        {
+            var settings = listPart?.GetSettings<ListPartSettings>();
+            if (settings != null)
+            {
+                if (contentItemType.Metadata.TryGetValue("ContainedContentTypes", out var containedContentTypes) &&
+                    containedContentTypes is IEnumerable<string> existingContainedContentTypes)
+                {
+                    contentItemType.Metadata["ContainedContentTypes"] = existingContainedContentTypes.Concat(settings.ContainedContentTypes).Distinct().ToArray();
+                }
+                else
```

Comment:
I personally would not include the else. I would just add a continue inside the TryGetValue block

Reply:
I may understand you, but if the metadata is not there, I must add it.